File: TextosJapones/src/main/java/org/jisho/textosJapones/database/dao/VocabularioDao.java
package org.jisho.textosJapones.database.dao;

import org.jisho.textosJapones.model.entities.Vocabulario;
import org.jisho.textosJapones.model.enums.Database;
import org.jisho.textosJapones.model.exceptions.ExcessaoBd;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;
import java.util.UUID;

public interface VocabularioDao {

	Database getTipo();

	void insert(Vocabulario obj) throws ExcessaoBd;

	void update(Vocabulario obj) throws ExcessaoBd;

	void delete(Vocabulario obj) throws ExcessaoBd;

	boolean exist(String vocabulario);

	Vocabulario select(String vocabulario, String base) throws ExcessaoBd;
	
	Vocabulario select(String vocabulario) throws ExcessaoBd;

	Vocabulario select(UUID id) throws ExcessaoBd;

	List<Vocabulario> selectAll() throws ExcessaoBd;
	
	void insertExclusao(String palavra) throws ExcessaoBd;
	
	boolean existeExclusao(String palavra, String basico) throws ExcessaoBd;
	
	Set<String> selectExclusao() throws ExcessaoBd;

	List<Vocabulario> selectEnvioVocabulario(LocalDateTime ultimo) throws ExcessaoBd;

	Set<String> selectExclusaoEnvio(LocalDateTime ultimo) throws ExcessaoBd;

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/database/dao/VocabularioDao.kt
package br.com.fenix.processatexto.database.dao

import br.com.fenix.processatexto.model.entities.processatexto.Vocabulario
import br.com.fenix.processatexto.model.enums.Database
import java.sql.SQLException
import java.time.LocalDateTime
import java.util.*


interface VocabularioDao : RepositoryDao<UUID?, Vocabulario> {
    val tipo: Database

    @Throws(SQLException::class)
    fun save(obj: Vocabulario)

    @Throws(SQLException::class)
    fun insertManual(obj: Vocabulario)

    @Throws(SQLException::class)
    fun updateManual(obj: Vocabulario)

    @Throws(SQLException::class)
    fun delete(obj: Vocabulario)

    fun exist(vocabulario: String): Boolean

    @Throws(SQLException::class)
    fun select(vocabulario: String, base: String): Optional<Vocabulario>

    @Throws(SQLException::class)
    fun select(vocabulario: String): Optional<Vocabulario>

    @Throws(SQLException::class)
    fun select(id: UUID?): Optional<Vocabulario>

    @Throws(SQLException::class)
    fun selectAll(): MutableList<Vocabulario>

    @Throws(SQLException::class)
    fun insertExclusao(palavra: String)

    @Throws(SQLException::class)
    fun existeExclusao(palavra: String, basico: String): Boolean

    @Throws(SQLException::class)
    fun selectExclusao(): Set<String>

    @Throws(SQLException::class)
    fun selectEnvioVocabulario(ultimo: LocalDateTime): List<Vocabulario>

    @Throws(SQLException::class)
    fun selectExclusaoEnvio(ultimo: LocalDateTime): List<String>
}